### PR TITLE
Show realnames on function's signature when enabled

### PR DESCRIPTION
**Pull request by ITAYC0HEN**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
This pull request will fix a problem in which the printed function signature did not honor the config `asm.flags.real`. 

In the following examples, notice how the name of the function changes from `sym.beet` to `beet` before and after the fix when `asm.flags.real` enabled.

**Before:**
```
[0x08048585]> e asm.flags.real=true
[0x08048585]> 
[0x08048585]> pdf
            ; CALL XREF from main @ 0x8048637
            ;-- eip:
╭ 112: sym.beet (char *src);
│           ; var char *s2 @ ebp-0x92
│           ; var int32_t var_8eh @ ebp-0x8e
│           ; var int32_t var_8ah @ ebp-0x8a
│           ; var char *dest @ ebp-0x88
│           ; arg char *src @ ebp+0x8
│           0x08048585      55             push ebp
│           0x08048586      89e5           mov ebp, esp
│           0x08048588      81ec98000000   sub esp, 0x98
│           0x0804858e      83ec08         sub esp, 8
```

**After**
```
[0x08048585]> e asm.flags.real=true
[0x08048585]> pdf
            ; CALL XREF from main @ 0x8048637
            ;-- eip:
╭ 112: beet (char *src);
│           ; var char *s2 @ ebp-0x92
│           ; var int32_t var_8eh @ ebp-0x8e
│           ; var int32_t var_8ah @ ebp-0x8a
│           ; var char *dest @ ebp-0x88
│           ; arg char *src @ ebp+0x8
│           0x08048585      55             push ebp
│           0x08048586      89e5           mov ebp, esp
│           0x08048588      81ec98000000   sub esp, 0x98
│           0x0804858e      83ec08         sub esp, 8

```

**Test plan**

- Open a binary that has realnames (e.g calls to imports in PE)
- Go to a function that has realname
- Print it with `asm.flags.real` disabled and see that the printed name is the non-realname
- Print it with `asm.flags.real` enabled and see that the printed name is the realname

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -1887,6 +1887,13 @@ static void ds_show_functions(RDisasmState *ds) {
 			r_cons_printf ("%s  ", COLOR_RESET (ds));
 		}
 		r_cons_printf ("%d: ", r_anal_function_realsize (f));
+
+		// show function's realname in the signature if realnames are enabled 
+		if (core->flags->realnames) {
+			RFlagItem *flag = r_flag_get (core->flags, fcn_name);
+			fcn_name = flag->realname;
```
Comment (radare) [inline libr/core/disasm.c:1894]: Possible null deref

File: libr/core/disasm.c [hunk 2]
```diff
@@ -1887,6 +1887,13 @@ static void ds_show_functions(RDisasmState *ds) {
 			r_cons_printf ("%s  ", COLOR_RESET (ds));
 		}
 		r_cons_printf ("%d: ", r_anal_function_realsize (f));
+
+		// show function's realname in the signature if realnames are enabled 
+		if (core->flags->realnames) {
+			RFlagItem *flag = r_flag_get (core->flags, fcn_name);
```
Comment (radare) [inline libr/core/disasm.c:1893]: Why assume a flag by name instead of offset? If the function is renamed this wont work

Comment (ITAYC0HEN) [reply libr/core/disasm.c:1893]: Aren't multiple flags possible for a single address? I see how it can become an issue

**Comment by radare** [comment]
Yes, and this should be the fallback imho. Or just use the rbin info instead of the flags becase rbin have the real name, the flags are just an imported version of it which can lead to inconsistencies if the user messes with it

File: libr/core/disasm.c [hunk 3]
```diff
@@ -1887,6 +1887,15 @@ static void ds_show_functions(RDisasmState *ds) {
...
+			if (flag->realname) {
```
Comment (radare) [inline libr/core/disasm.c:1894]: if (flag &&  flag->realname)

**Comment by ITAYC0HEN** [comment]
will fix the test tomorrow :)

**Comment by radare** [comment]
👍

File: libr/core/disasm.c (hunk 2 above)
Comment (thestr4ng3r) [reply libr/core/disasm.c:1893]: Name is the correct approach. When a function is renamed, the flag is renamed in most cases as well. If there is a flag that has a different name, it is not the function's anymore. Also names uniquely identify flags while addresses don't.

**Pull request merged**